language/move-lang/tests/move_check/borrows/borrow_global_invalid.exp

```diff
@@ -1,77 +1,56 @@
+
+error[E07001]: referential transparency violated
+   ┌─ tests/move_check/borrows/borrow_global_invalid.move:12:18
    │
- 6 │         let r2 = borrow_global<R>(addr);
+11 │         let f = &mut borrow_global_mut<R>(addr).f;
+   │                 --------------------------------- It is still being mutably borrowed by this reference
+12 │         let r2 = borrow_global<R>(addr);
    │                  ^^^^^^^^^^^^^^^^^^^^^^ Invalid borrowing of resource 'R'
-   ·
- 5 │         let r1 = borrow_global_mut<R>(addr);
-   │                  -------------------------- It is still being mutably borrowed by this reference
-   │
-
-error: 
-
-    ┌── tests/move_check/borrows/borrow_global_invalid.move:12:18 ───
-    │
- 12 │         let r2 = borrow_global<R>(addr);
-    │                  ^^^^^^^^^^^^^^^^^^^^^^ Invalid borrowing of resource 'R'
-    ·
- 11 │         let f = &mut borrow_global_mut<R>(addr).f;
-    │                 --------------------------------- It is still being mutably borrowed by this reference
-    │
-
-error: 
 
-    ┌── tests/move_check/borrows/borrow_global_invalid.move:18:18 ───
-    │
- 18 │         let f = &borrow_global<R>(addr).f;
-    │                  ^^^^^^^^^^^^^^^^^^^^^^ Invalid borrowing of resource 'R'
-    ·
- 17 │         let r1 = borrow_global_mut<R>(addr);
-    │                  -------------------------- It is still being mutably borrowed by this reference
-    │
-
-error: 
-
-    ┌── tests/move_check/borrows/borrow_global_invalid.move:24:18 ───
-    │
- 24 │         let r1 = borrow_global_mut<R>(addr);
-    │                  ^^^^^^^^^^^^^^^^^^^^^^^^^^ Invalid borrowing of resource 'R'
-    ·
- 23 │         let r2 = borrow_global<R>(addr);
-    │                  ---------------------- It is still being borrowed by this reference
-    │
-
-error: 
-
-    ┌── tests/move_check/borrows/borrow_global_invalid.move:30:18 ───
-    │
- 30 │         let r2 = borrow_global<R>(addr);
-    │                  ^^^^^^^^^^^^^^^^^^^^^^ Invalid borrowing of resource 'R'
-    ·
- 29 │         let f = &mut borrow_global_mut<R>(addr).f;
-    │                 --------------------------------- It is still being mutably borrowed by this reference
-    │
+error[E07001]: referential transparency violated
+   ┌─ tests/move_check/borrows/borrow_global_invalid.move:18:18
+   │
+17 │         let r1 = borrow_global_mut<R>(addr);
+   │                  -------------------------- It is still being mutably borrowed by this reference
+18 │         let f = &borrow_global<R>(addr).f;
+   │                  ^^^^^^^^^^^^^^^^^^^^^^ Invalid borrowing of resource 'R'
 
-error: 
+error[E07002]: mutable ownership violated
+   ┌─ tests/move_check/borrows/borrow_global_invalid.move:24:18
+   │
+23 │         let r2 = borrow_global<R>(addr);
+   │                  ---------------------- It is still being borrowed by this reference
+24 │         let r1 = borrow_global_mut<R>(addr);
```

Comment:
Should this first error be pointing to the `borrow_global` or to `r2`? With the error message I feel as though `r2` is the correct placement of the location as `r2` is the reference that is still live on line 24. I can also see the reasoning behind pointing to the `borrow_global` -- in which case what would you think of pointing to the whole assignment; `r2 = borrow_global<R>(addr)`.

Reply:
I don't think there is anything wrong with the locs.
The primary label (the ^^^^^^^ one) is pointing to where the error is happening.
The secondary label (the -------- one) is pointing to the live reference that is causing the error. The borrow checker only remembers where references where created, not where they currently live. It could be interesting to maybe try to provide that information too but that is far beyond the scope of this PR